### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by harsh-tamr**
Update all dependencies, and fix the onSelect behavior

### CODE REVIEW

**Comment by harsh-tamr** [comment]
Fixes https://github.com/uptick/react-keyed-file-browser/issues/115 and also adds a story in the storybook

**Comment by harsh-tamr** [comment]
@bor3ham @scaredcat @roymfibo @jlo-1

**Comment by jarekwg** [comment]
Hey @harsh-tamr,

Thanks for your contribution!
We're all quite under the pump atm, but promise to take a look soon.

Cheers,

File: stories/index.js [hunk 1]
```diff
@@ -94,6 +99,29 @@ export const differentRendersAndGroupers = () => (
...
 )
 
+export const selectHandler = () => (
+  <FileBrowser
+    icons={Icons.FontAwesome(4)}
+    files={`a/,0,0
+a/b/,0,0
+a/b/c-1.csv,1596791355222,383
+a/b/c-2.csv,1596791353289,363
+x/y/z/broken.csv,1598519377165,323`
```
Comment (mwfister) [inline stories/index.js:109]: I imagine this story may have been used to help troubleshooting? I think we can roll it into the other stories. Thanks to your changes, we can update `files` to remove the explicit folder (first entry) and apply an `onSelect` to one of the stories.

```
const files = [
  {
    key: 'animals/',
    modified: addHours(new Date(), 1).getTime(),
    size: 0,
  },
  {
    key: 'animals/dog.png',
    modified: subHours(new Date(), 1).getTime(),
    size: 0,
  },
```

File: package.json [hunk 2]
```diff
@@ -33,48 +36,49 @@
...
-    "webpack": "^4.29.1",
-    "webpack-cli": "^3.2.3",
-    "webpack-node-externals": "^1.5.4"
+    "node-sass": "^4.14.1",
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1",
+    "sass-loader": "^10.0.2",
+    "style-loader": "^1.2.1",
+    "typescript": "^4.0.2",
+    "url-loader": "^4.1.0",
```
Comment (mwfister) [inline package.json:79]: What is this `url-loader` being used for?

File: webpack.config.js [hunk 3]
```diff
@@ -13,12 +13,12 @@ module.exports = {
...
       {
         test: /\.(js|jsx)$/,
-        loader: ['babel-loader'],
+        loader: [require.resolve('babel-loader')],
         include: path.join(__dirname, './src/'),
       },
       {
         test: /.*\.sass*/,
-        loader: ['style-loader', 'css-loader', 'sass-loader'],
+        loader: [require.resolve('style-loader'), require.resolve('css-loader'), require.resolve('sass-loader')],
```
Comment (mwfister) [inline webpack.config.js:21]: I'm not too familiar with webpack intricacies, what's the idea behind this change?

File: package.json [hunk 4]
```diff
@@ -33,48 +36,49 @@
   "homepage": "https://github.com/uptick/react-keyed-file-browser#readme",
   "dependencies": {
```
Comment (mwfister) [inline package.json:37]: These loaders have a minimum required node version, we should probably be specifying something to that effect
```suggestion
  "engines": {
    "node": ">=10.13.0",
  },
  "dependencies": {
```

**Review by mwfister — CHANGES_REQUESTED** [review]
Thanks for your help improving this package 👍. The getFile change looks good, so I'll approve that in the separate PR. I have a couple of questions/suggestions for the dependency work below.

File: stories/index.js (hunk 1 above)
Comment (harsh-tamr) [reply stories/index.js:109]: sounds good

File: package.json (hunk 2 above)
Comment (harsh-tamr) [reply package.json:82]: I think it was a transitive dependency, I'll look into it

File: webpack.config.js (hunk 3 above)
Comment (harsh-tamr) [reply webpack.config.js:21]: https://webpack.js.org/api/module-methods/#requireresolve

I was playing around with yarn pnp and I ran into some issues https://classic.yarnpkg.com/en/docs/pnp/troubleshooting/ It's a way of future proofing. I'm not attached to it and can take it out

Comment (mwfister) [reply webpack.config.js:21]: That's fine to keep. I was curious about it, but not particularly opinionated.

**Comment by harsh-tamr** [comment]
sorry got super busy, but will take a look soon

File: package.json [hunk 5]
```diff
@@ -23,58 +23,65 @@
...
+    "gcs"
   ],
   "author": "Uptick Pty Ltd",
   "license": "MIT",
   "bugs": {
     "url": "https://github.com/uptick/react-keyed-file-browser/issues"
   },
   "homepage": "https://github.com/uptick/react-keyed-file-browser#readme",
+  "engines": {
+    "node": ">=10.13.0",
```
Comment (dankolbman) [inline package.json:38]: Trailing comma needs to be removed.

**Review by dankolbman — COMMENTED** [review]
Thanks for this PR! The deprecation of `DragDropContext` is preventing us from using the master branch in production.
I was able to get things working by using the first commit of this branch, although I had to manually build and install the package to do so.

File: package.json (hunk 5 above)
Comment (harsh-tamr) [reply package.json]: resolved just now

**Comment by dreadera** [comment]
Closing This PR as we have updated our dependencies.

**Pull request closed without merging**